### PR TITLE
add ai2 footer to lib and use it in the demo

### PR DESCRIPTION
**Pull request by jonborchardt**
Add basic footer to lib and demo site and remove existing
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -259,7 +260,9 @@
   },
   "peerDependencies": {
     "react": ">=16.8.0",
-    "react-dom": ">=16.8.0"
+    "react-dom": ">=16.8.0",
+    "react-router-dom": "^5.0.1",
```
Comment (codeviking) [inline package.json]: Are we sure we want to add a hard-dependency on this? This created problems for NextJS builds in the past.  Maybe we omit this and try to build away from that dependency interlinking now?

File: components/link/InternalLink.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import styled from 'styled-components';
+import { NavLink } from 'react-router-dom';
```
Comment (codeviking) [inline components/link/InternalLink.tsx]: We should try to opt-out of `react-router` this time around. It'll save us time on NextJS based projects, which seems to be "the way" we're headed right now :).

File: components/layout/LayoutContext.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        <LayoutContext.Provider
+            value={{
+                layoutVariant,
+                currentHeaderHeight,
+                setHeaderHeight,
+            }}>
+            {children}
+        </LayoutContext.Provider>
+    );
+};
```
Comment (codeviking) [inline components/layout/LayoutContext.tsx]: Is there a `Layout` mechanism provided by `antd` that we should use instead, maybe?

File: components/link/InternalLink.tsx (hunk 2 above)
Comment (jonborchardt) [reply components/link/InternalLink.tsx]: removed

File: components/layout/LayoutContext.tsx (hunk 3 above)
Comment (jonborchardt) [reply components/layout/LayoutContext.tsx]: https://github.com/allenai/ant-design/issues/31

**Pull request merged**